### PR TITLE
handle the line breaks in link cells

### LINKED ISSUE #10078: "\n" chars in a cell value must be copied to clipboard and be displayed as new line in results grid
**Feature descritpion**
In SELECT clause results, if a field/column text value contains "\n" characters, the cell value at least should be copied to clipboard with these "\n" characters, at best - be shown as multi-line text in the cell.

**Copy to Clipboard**
This is how Azure Data Studio shows such a value:
image

When I copy the cell value:
image

and paste it in a plain text editor, I see this:
image

whereas the correctly copied value would be:
image

**Show the value in SELECT results**
E.g. this is how Azure Data Studio shows such a value:
image

and this is how DataGrip shows the same value (much more convenient, in my opinion):
image

After the cell is bouble-clicked in DataGrip:
image

### PR DESCRIPTION
**Pull request by alanrenmsft**
this PR fixes #10078

1. apply the line breaking and maximum length logic used in text cells to link cells
2. display line break characters.

before:
image

after:
image

### CODE REVIEW

File: src/sql/base/browser/ui/table/formatters.ts [hunk 1]
```diff
@@ -71,13 +71,14 @@ export function hyperLinkFormatter(row: number | undefined, cell: any | undefine
...
 			cellClasses += ' xmlLink';
-			valueToDisplay = escape(value.displayValue);
+			valueToDisplay = getCellDisplayValue(value.displayValue);
 			return `<a class="${cellClasses}">${valueToDisplay}</a>`;
 		} else {
 			cellClasses += ' missing-value';
 		}
 	} else if (isHyperlinkCellValue(value)) {
-		return `<a class="${cellClasses}" title="${escape(value.displayText)}">${escape(value.displayText)}</a>`;
+		valueToDisplay = getCellDisplayValue(value.displayText);
```
Comment (kburtram) [inline src/sql/base/browser/ui/table/formatters.ts:80]: I assume truncating doesn't change the behavior when clicking the link.  i.e. the value in the new document comes from the underlying data model, not the cell.

File: src/sql/base/browser/ui/table/formatters.ts [hunk 2]
```diff
@@ -119,13 +119,18 @@ export function textFormatter(row: number | undefined, cell: any | undefined, va
...
-		valueToDisplay = escape(valueToDisplay.length > 250 ? valueToDisplay.slice(0, 250) + '...' : valueToDisplay);
+		valueToDisplay = getCellDisplayValue(valueToDisplay);
 		titleValue = valueToDisplay;
 	}
 
 	return `<span title="${titleValue}" style="${cellStyle}" class="${cellClasses}">${valueToDisplay}</span>`;
 }
 
+function getCellDisplayValue(cellValue: string): string {
+	let valueToDisplay = cellValue.replace(/(\r\n|\n|\r)/g, ' ');
```
Comment (kburtram) [inline src/sql/base/browser/ui/table/formatters.ts:130]: I'm probably missing something but this appears to replace the newlines with a space.  Where are the visible linebreak characters in the screenshot coming from?

**Review by kburtram — APPROVED** [review]

File: src/sql/base/browser/ui/table/formatters.ts (hunk 1 above)
Comment (alanrenmsft) [reply src/sql/base/browser/ui/table/formatters.ts:80]: yes, that is correct.

File: src/sql/base/browser/ui/table/formatters.ts (hunk 2 above)
Comment (alanrenmsft) [reply src/sql/base/browser/ui/table/formatters.ts]: good catch, I forgot to push the last commit. done now.

**Review by lewis-sanchez — APPROVED** [review]

**Pull request merged**